common/common-rest/src/main/java/org/apache/servicecomb/common/rest/AbstractRestInvocation.java

```diff
@@ -103,7 +103,7 @@ protected void setContext() throws Exception {
     @SuppressWarnings("unchecked")
     Map<String, String> cseContext =
         JsonUtils.readValue(strCseContext.getBytes(StandardCharsets.UTF_8), Map.class);
-    invocation.setContext(cseContext);
+    invocation.addContext(cseContext);
```

Comment:
highway have the same problem

we can add a new method in invocation: mergeContext
1.if new context have more items, then addAll to new context, and replace old context
2.if new context have less items, then allAll to old context directly.

Reply:
done